### PR TITLE
Agent

### PR DESCRIPTION
**Pull request by ADimoska**

### CODE REVIEW

File: pkg/infra/server/baseServer.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	NumAgents int
+	NumTurns  int
+	Agents    []Agent
+}
+
+type Agent struct {
+	ID uuid.UUID
+	name string
+	floor int
+	energy int
```
Comment (MattSScott) [inline pkg/infra/server/baseServer.go]: This is too specific - I think that this came from a previous Cohort. I'd remove them to keep the implementation general

File: pkg/infra/server/baseServer.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		ID: uuid.New(),
+		name: name,
+		floor: floor,
+		energy: energy,
+	}
+}  
+
+func ReduceEnergy(a *Agent){
+	a.energy-=10
+}
```
Comment (MattSScott) [inline pkg/infra/server/baseServer.go]: Same as before

File: pkg/infra/server/baseServer.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+func (bs *BaseServer) RunGameLoop(loopnum int) {
+	fmt.Printf("Game Loop %d Running \n", loopnum)
+	fmt.Printf("%d agents initialised: \n", bs.NumAgents)
+	for index, element := range bs.Agents {
+		//fmt.Printf("agent %d has id: %s \n", index, element.ID)
+		fmt.Printf("agent %d has name: %s \n", index, element.name)
+		//fmt.Printf("agent %d has name: %d \n", index, element.floor)
+		fmt.Printf("agent %d has energy: %d \n", index, element.energy)
+		fmt.Printf("_____________________________________________ \n")
+		bs.Agents[index].UpdateAgent()
```
Comment (MattSScott) [inline pkg/infra/server/baseServer.go:40]: Use the `element` variable you just created. Also, maybe label it as `agent`

**Review by MattSScott — CHANGES_REQUESTED** [review]
Really nice first implementation. Just a few changes to make

File: pkg/agents/baseAgent.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	//"fmt"
+	//"strconv"
+	"github.com/google/uuid"
+)
+
+type Agent struct {
+	ID uuid.UUID
+	Name string
+	Floor int
+	Energy int
```
Comment (MattSScott) [inline pkg/agents/baseAgent.go:13]: Please remove these - they are too specific

File: pkg/agents/baseAgent.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}  
+
+func ReduceEnergy(a *Agent){
+	a.Energy-=10
+}
+
+func (a *Agent)UpdateAgent(){
+		ReduceEnergy(a)
+		
+	}
```
Comment (MattSScott) [inline pkg/agents/baseAgent.go:32]: Same as above

**Review by MattSScott — CHANGES_REQUESTED** [review]
Again, good stuff, but a few parameters that are implementation specific that need removing

**Comment by ADimoska** [comment]
Stale - new pull request submitted

**Pull request closed without merging**